src/mpi/init/init_bindings.c

```diff
@@ -6,36 +6,6 @@
 #include "mpiimpl.h"
 #include "mpi_init.h"
 
-/* ** FORTRAN binding **************/
-#if defined(HAVE_FORTRAN_BINDING) && defined(HAVE_MPI_F_INIT_WORKS_WITH_C)
-#ifdef F77_NAME_UPPER
-#define mpirinitf_ MPIRINITF
-#elif defined(F77_NAME_LOWER) || defined(F77_NAME_MIXED)
-#define mpirinitf_ mpirinitf
-#endif
-void mpirinitf_(void);
-
-void MPII_init_binding_fortran(void)
-{
-    /* Initialize the C versions of the Fortran link-time constants.
-     *
-     * We now initialize the Fortran symbols from within the Fortran
-     * interface in the routine that first needs the symbols.
-     * This fixes a problem with symbols added by a Fortran compiler that
-     * are not part of the C runtime environment (the Portland group
-     * compilers would do this)
-     */
-    mpirinitf_();
```

Comment:
Is there ever a case where C needs this information and we can't rely on Fortran initialization? Inter-language support? I just asking, I don't know for sure.

Reply:
`mpirinitf_()` is only to tell C what address constants such as `MPI_BOTTOM`, `MPI_IN_PLACE`, etc. FORTRAN will be using. We only need it if there is FORTRAN code. There is a lazy mechanism calling this in the fortran binding. Reference `MPIR_F_NeedInit`.